internal/reference/reference.go

```diff
@@ -139,6 +143,10 @@ func (r recordReference) Delete() error {
 	return fmt.Errorf("cannot delete record: %w", ErrImmutableReference)
 }
 
+func (r recordReference) Rename(string) error {
+	return fmt.Errorf("cannot rename record: %w", ErrImmutableReference)
```

Comment:
`ErrImmutableReference` says "cannot set immutable reference", that doesn't seem right here. I'd rather create a separate error for when renaming is not allowed.